sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -64,6 +66,7 @@ public OfferEvaluator(StateStore stateStore, OfferRequirementProvider offerRequi
             }
 
             if (!failureNotifications.isEmpty()) {
+                recommendations.clear();
```

Comment:
This line comes from an abundance of caution.  It's a variable derived from the recommendationSlate, which is recreated anew on ever pass of the loop.  In fact the variable could probably go away altogether, let me look at that.